src/store/proposals/sagas/addProposal.test.ts

```diff
@@ -0,0 +1,557 @@
+      payload: {
+        SendProposalContract: GovernorTypes.GovernorAdmin,
+        Description: 'test description of proposal',
+        Values: [
+          {
+            Target: '0x03f4',
+            Value: '1000000000',
+            Signature: 'Signature',
+            Calldata: '0x616263640000',
+          },
+        ],
+      },
+      type: 'proposals/addProposal',
+    };
+
+    const getBasePath = () =>
+      expectSaga(addProposal, proposalGovernorAdminPayload)
+        .withReducer(reducer)
+        .withState(initialState)
+        .select(accountSelector);
+
+    it('happy path', async () => {
+      await getBasePath()
+        .provide([
+          [matchers.select(accountSelector), testAccount],
+          [matchers.select(governorAdminSelector), mockGovernorContract],
+          [
+            matchers.call(
+              mockGovernorContract.propose,
+              [proposalGovernorAdminPayload.payload.Values[0].Target],
+              [
+                proposalGovernorAdminPayload.payload.Values[0].Value,
+              ] as BigNumberish[],
+              [proposalGovernorAdminPayload.payload.Values[0].Signature],
+              [proposalGovernorAdminPayload.payload.Values[0].Calldata],
+              proposalGovernorAdminPayload.payload.Description
+            ),
+            mockTx,
+          ],
+          [matchers.call(mockTx.wait), mockTxReceipt],
+        ])
+        .select(governorAdminSelector)
+        .call(
+          mockGovernorContract.propose,
+          [proposalGovernorAdminPayload.payload.Values[0].Target],
+          [proposalGovernorAdminPayload.payload.Values[0].Value],
+          [proposalGovernorAdminPayload.payload.Values[0].Signature],
+          [proposalGovernorAdminPayload.payload.Values[0].Calldata],
+          proposalGovernorAdminPayload.payload.Description
+        )
+        .hasFinalState(successState)
+        .run();
+    });
+
+    it('when wallet is not connected', async () => {
+      await getBasePath()
+        .provide([
+          [matchers.select(accountSelector), undefined],
+          [matchers.select(governorAdminSelector), mockGovernorContract],
+          [
+            matchers.call(
+              mockGovernorContract.propose,
+              [proposalGovernorAdminPayload.payload.Values[0].Target],
+              [
+                proposalGovernorAdminPayload.payload.Values[0].Value,
+              ] as BigNumberish[],
+              [proposalGovernorAdminPayload.payload.Values[0].Signature],
+              [proposalGovernorAdminPayload.payload.Values[0].Calldata],
+              proposalGovernorAdminPayload.payload.Description
+            ),
+            mockTx,
+          ],
+          [matchers.call(mockTx.wait), mockTxReceipt],
+        ])
+        .select(governorAdminSelector)
+        .hasFinalState(failureWalletNotConnectedState)
+        .run();
+    });
+
+    it('fetching error', async () => {
+      await getBasePath()
```

Comment:
I would add a check to verify that saga is stopping in the proper time and is not yielding any redundant effects.

```suggestion
      const runResult = await getBasePath()
      
      expect(runResult.effects).toEqual({})
```